### ISSUE TITLE
GPU input for `XGBClassifier`.

### ISSUE THREAD

**Issue by trivialfis** [dmlc/xgboost#5541]
Currently the `XGBClassifier` employs `numpy.unique` and `sklearn.LabelEncoder` for determining number of classes and encoding target variables.  But the workflow breaks GPU inputs like `cupy` or `cudf`.

**Comment by hcho3** [comment]
So anyone who wants to use GPU have to use the "traditional" learning API (`xgb.train()` etc)?

**Comment by trivialfis** [comment]
For classification, yup.

**Comment by trivialfis** [comment]
Use GPU data structure as input, like `cudf` is effected.  But training with XGBoost `GPU Hist` is not effected.

**Comment by trivialfis** [comment]
@hcho3  How does `cuml` handle these kind of things?  I'm thinking about adding `num_class` as a parameter for `XGBClassifier`, and possibly enforce this parameter to be set in the future.  The label encoder is not ideal anyway.

**Comment by hcho3** [comment]
@trivialfis The random forest classifier uses `unique()` to determine the number of classes: https://github.com/rapidsai/cuml/blob/4b3d2be37c6ee18330bdf9091588248c5e8ac5d5/python/cuml/dask/ensemble/randomforestclassifier.py#L395. The assumption is that the labels are integers between 0 and `num_class - 1`.

**Comment by trivialfis** [comment]
@hcho3  Thanks for the input.

**Comment by trivialfis** [comment]
Closing in favor of https://github.com/dmlc/xgboost/issues/6232 .
